zcash_client_backend/src/data_api.rs

```diff
@@ -321,6 +321,17 @@ impl AccountBalance {
     }
 }
 
+/// An enumeration used to control what information is tracked by the wallet for
+/// notes received by a given account.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub enum AccountType {
```

Comment:
Naming is hard, but I don't like the name `AccountType`. We went through a similar discussion before to arrive at `AccountSource`, which is "the kinds of accounts supported". Kind and type are synonyms to users, so this new `AccountType` confuses the public API.

I think a better name would be something like `AccountUsage / AccountUse`, as this enum defines how the account can be used by the wallet user (due to it influencing what the wallet tracks). I also wondered about `AccountPurpose`, but that feels more inherent to the account, whereas this enum is about the account in the context of this particular wallet.

Reply:
`AccountPurpose` seems good to me, let's go with that; we're talking about the wallet account, not the ZIP 32 account, and in the context of the wallet this is indeed what the purpose is for having the account in the wallet.